Alliance.Client/Extensions/AdminMenu/ViewModels/AdminVM.cs

```diff
@@ -1,491 +1,513 @@
+		public string Score
+		{
+			get
+			{
+				return _Score;
+			}
+			set
+			{
+				if (value != _Score)
+				{
+					_Score = value;
+					OnPropertyChangedWithValue(value, "Score");
+				}
+			}
+		}
+
+		[DataSourceProperty]
+		public string FilterText
+		{
+			get
+			{
+				return _filterText;
+			}
+			set
+			{
+				if (value != _filterText)
+				{
+					_filterText = value;
+					FilterPlayers(_filterText);
+					OnPropertyChangedWithValue(value, "FilterText");
+				}
+			}
+		}
+
+		/// <summary>
+		/// Filter list of players with given text filter
+		/// </summary>
+		public void FilterPlayers(string filterText)
+		{
+			foreach (NetworkPeerVM networkPeerVM in _networkCommunicators)
+			{
+				if (networkPeerVM != null)
+				{
+					networkPeerVM.IsFiltered = !networkPeerVM.Username.ToLower().Contains(filterText);
+				}
+			}
+		}
+
+		public void Heal()
+		{
+			if (_selectedPeer == null) { return; }
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { Heal = true, PlayerSelected = _selectedPeer.PeerId });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void HealAll()
+		{
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { HealAll = true, PlayerSelected = null });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void GodMod()
+		{
+			if (_selectedPeer == null) { return; }
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { GodMod = true, PlayerSelected = _selectedPeer.PeerId });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void GodModAll()
+		{
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { GodModAll = true, PlayerSelected = null });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void KillPlayer()
+		{
+			if (_selectedPeer == null) { return; }
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { Kill = true, PlayerSelected = _selectedPeer.PeerId });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void KillAll()
+		{
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { KillAll = true, PlayerSelected =  null });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void KickPlayer()
+		{
+			if (_selectedPeer == null) { return; }
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { Kick = true, PlayerSelected = _selectedPeer.PeerId });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void SendWarningToPlayer(string customWarning)
+		{
+			if (_selectedPeer == null) { return; }
+			GameNetwork.BeginModuleEventAsClient();
+			GameNetwork.WriteMessage(new AdminClient() { SendWarningToPlayer = true, PlayerSelected = _selectedPeer.PeerId, WarningMessageToPlayer = customWarning });
+			GameNetwork.EndModuleEventAsClient();
+		}
+
+		public void CustomWarning()
```

Comment:
I would rename this method to something like `PrompWarningMessageSelection`

Reply:
Done into AdminVM.cs & AdminPanel.xml